docs/source/user/jitclass.rst

```diff
@@ -40,11 +40,49 @@ initializing each defined fields.  Uninitialized fields contains garbage data.
 Methods and properties (getters and setters only) can be defined.  They will be
 automatically compiled.
 
-Specifying ``numba.typed`` containers as class members
-======================================================
-It is often desirable to use a ``numba.typed.Dict`` or a ``numba.typed.List`` as
-a class member in a ``jitclass``. Methods for using these types and various
-common patterns are presented in the following:
+
+Inferred class member types with ``as_numba_type``
+==================================================
+
+Fields of a ``jitclass`` can also be inferred from Python type annotations.
+
+.. literalinclude:: ../../../numba/tests/doc_examples/test_jitclass.py
+   :language: python
+   :start-after: magictoken.ex_jitclass_type_hints.begin
+   :end-before: magictoken.ex_jitclass_type_hints.end
+   :dedent: 8
+
+Any type annotations on the class will be used to extend the spec if that field
+is not already present.  The Numba type corresponding to given Python type is
+inferred using ``as_numba_type``.  For example, if we have the class
```

Comment:
Perhaps mention that type annotations on method arguments and return type are ignored since they are used in the example.

Reply:
Good call.  Done.